src/app/execution.rs

```diff
@@ -122,35 +126,100 @@ impl ExecutionContext {
+                            match maybe_batch {
+                                Ok(batch) => {
+                                    batches.push(batch);
+                                }
+                                Err(e) => {
+                                    let elapsed = start.elapsed();
+                                    query.set_error(Some(e.to_string()));
+                                    query.set_execution_time(elapsed);
+                                    break;
+                                }
+                            }
+                        }
+                        let elapsed = start.elapsed();
+                        let rows: usize = batches.iter().map(|r| r.num_rows()).sum();
+                        query.set_results(Some(batches));
+                        query.set_num_rows(Some(rows));
+                        query.set_execution_time(elapsed);
+                    }
+                } else {
+                    info!("Executing query and discarding results");
+                    self.execute_sql(sql, false).await?;
+                    let elapsed = start.elapsed();
+                    query.set_execution_time(elapsed);
+                }
+            } else {
+                self.execute_sql(sql, false).await?;
+                let elapsed = start.elapsed();
+                query.set_execution_time(elapsed);
+            }
+            _sender.send(AppEvent::QueryResult(query)).unwrap();
+        }
+        Ok(())
+    }
+
+    /// Execcutes the specified parsed DataFusion statement and discards the result
+    pub async fn execute_sql(&self, sql: &str, print: bool) -> Result<()> {
+        let df = self.session_ctx.sql(sql).await?;
+        self.execute_stream_dataframe(df, print).await
+    }
+
     /// Executes the specified parsed DataFusion statement and prints the result to stdout
-    pub async fn execute_statement(&self, statement: Statement) -> Result<()> {
+    pub async fn execute_and_print_statement(&self, statement: Statement) -> Result<()> {
         let plan = self
             .session_ctx
             .state()
             .statement_to_plan(statement)
             .await?;
         let df = self.session_ctx.execute_logical_plan(plan).await?;
-        self.execute_stream_dataframe(df).await
+        self.execute_stream_dataframe(df, true).await
     }
 
     /// Executes the specified query and prints the result to stdout
-    pub async fn execute_stream_sql(&self, query: &str) -> Result<()> {
+    pub async fn execute_and_print_stream_sql(&self, query: &str) -> Result<()> {
         let df = self.session_ctx.sql(query).await?;
-        self.execute_stream_dataframe(df).await
+        self.execute_stream_dataframe(df, true).await
     }
 
-    pub async fn execute_stream_dataframe(&self, df: DataFrame) -> Result<()> {
+    pub async fn execute_stream_dataframe(&self, df: DataFrame, print: bool) -> Result<()> {
```

Comment:
maybe we should name this `execute_and_print_stream_dataframe` too

Reply:
It might not print though if `print` is `false`. The ones i renamed to have `print` in the name i call `execute_stream_dataframe` with `print` set to `true`